docs/hal/flash.md

```diff
@@ -0,0 +1,95 @@
+# FLASH IAP porting guide
+
+Flash IAP set-by-step list:
+
+1. Update target to support bootloader
+    * Update linker script
+    * Implement flash HAL API
+    * Add pack support to mbed-os
```

Comment:
Russ is working currently on adding this to the mbed-os. I can remove it for now, and add it later. 

@AnotherButler If you do any edits, please remove this, will be added with more information later